#PART 1:

**TASKS:**
<br>
*1. Derive the null and alternative hypothesis.* <br>
*2. Determine whether the test is right-tailed or left-tailed*

1. Statement:
A manufacturer states that the percentage of defective items produced by a new machine is less than the industry average of 5%.

Null(H0): p>= 0.05 = % of defective items by new machine >= industry average of 5% and p is proportion of defective items
<br>
Alternative(H1): p<0.05 = test % of defective items are less than average 5%


Test: Since alternative states p < 5% it's 1 tailed test - left-tailed

2. Statement:
An educational consultant suggests that the average test scores of students who attend a test preparation course are greater than the national average.

Null:(H0): population mean of test scores who attend <= national average test score (no effect hypothesis)
<br>
Alternative(H1): population mean of test scores who attend > national average test score

Test: Testing alternative > indicates 1-tailed test, a right-tailed test.

#PART 2:

Use scipy.stats for this. All 5 steps should be seen.

1. A school claimed that the students who study that are more intelligent than the average school. On calculating the IQ scores of 50 students, the average turns out to be 110. The mean of the population IQ is 100 and the standard deviation is 15. State whether the claim of the principal is right or not at a 5% significance level.



In [ ]:
import numpy as np
import scipy.stats as stats

**Step 1: Formulate research question and hypothesis:**

Null(H0): no effect-students who study are <= average school intelligence

Alternate(H1): test claim-students who study are > average school intelligence

Right-tailed

**Step 2: Collect a sample and calculate the sample mean (x bar: x̄) and standard deviation (s)**

xˉ is the sample mean (110)

μ is the population mean (100)

σ is the population standard deviation (15)

n is the sample size (50)

z = 110 - 100 / 15 squareroot 50

z = 10 /(15/7.0710)= 10/2.12 = 4.72

In [ ]:
# Step 3: Use the one-tailed z-test formula to calculate the test statistic.
from scipy.stats import norm
sample_mean = 110
population_mean = 100
population_std = 15
sample_size = 50
alpha = 0.05

# Find critical value(z_alpha)
z_alpha = norm.ppf(1 - alpha)
print("Critical value (z_alpha):", z_alpha)

z_score = (sample_mean - population_mean) / (population_std / np.sqrt(sample_size))
print("Z-Score:", z_score)

Critical value (z_alpha): 1.6448536269514722
Z-Score: 4.714045207910317


4.72 > 1.64 we reject null hypothesis

4.71 Z-Score Sample mean is 4.71 standard deviations above population mean. This suggests sample mean of 110 is higher than population mean of 100, providing evidence to support claim.

In [ ]:
# Step 4: find the p-value.
from scipy import stats

# Calculate the p-value
p_value = 1 - norm.cdf(z_score)

print('P-value:', p_value)

P-value: 1.2142337364462463e-06


P-Value is 1.2142 X 10 to power of -6 which is equivilant to  0.0000012142337364462463.

P-Value is smaller than 0.05 which is evidence against  null hypotheses.

In [ ]:
# Step 5: Reject or fail to reject the null hypothesis based on the significance level.
print('p-value :',p_value)

# Hypothesis
if p_value <  alpha:
    print("Reject Null Hypothesis")
else:
  print("Fail to Reject Null Hypothesis")

p-value : 1.2142337364462463e-06
Reject Null Hypothesis


2. A company claims that the average response time for their customer support hotline is less than 5 minutes. You conducted a sample of 36 calls and found that the average response time was 4.7 minutes with a standard deviation of 1.2 minutes. Conduct a hypothesis test to determine whether there is sufficient evidence to support the company's claim.

**Step 1: Formulate research question and hypothesis:**

Null(H0): no effect-average response time for cs hotline >= 5 minutes

Alternative(H1): test claim-average response time for cs hotline < 5 minutes

Left-tailed

**Step 2: Collect a sample and calculate the sample mean (x bar: x̄) and standard deviation (s)**

xˉ is the sample mean (4.7)

μ is the population mean (5)

σ is the population standard deviation (1.2)

n is the sample size (36)

z = 4.7 - 5 / 1.2 / squareroot 36

z = -0.3 /(1.2/6)= -0.3/.2 = -0.15


In [ ]:
# Step 3: Use the one-tailed z-test formula to calculate the test statistic.
from scipy.stats import norm
sample_mean = 4.7
population_mean = 5
population_std = 1.2
sample_size = 36
alpha = 0.05

# Find critical value(z_alpha)
z_alpha = norm.ppf(1 - alpha)
print("Critical value (z_alpha):", z_alpha)

z_score = (sample_mean - population_mean) / (population_std / np.sqrt(sample_size))
print("Z-Score:", z_score)

Critical value (z_alpha): 1.6448536269514722
Z-Score: -1.4999999999999993


-1.5 < 1.64 we accept null hypothesis (Zscore < critical value)

-1.5 Z-Score shows sample mean is 1.5 sd below population mean. Not enough to support claim.

In [ ]:
# Step 4: find the p-value.
from scipy import stats

# Calculate the p-value cumulative distribution function (CDF) since we're testing for less than (1-CDF)
p_value = norm.cdf(z_score)

print('P-value:', p_value)


P-value: 0.06680720126885814


P-Value is 0.06680720126885814

P-Value is greater than 0.05 which is evidence against the alternative hypotheses and cannot reject null hypotheses. There's around a 6.68% probability of observing the sample mean of 4.7 minutes or less. There's not enough evidence to support alternative hypotheses.

In [ ]:
# Step 5: Reject or fail to reject the null hypothesis based on the significance level.
print('p-value :',p_value)

# Hypothesis
if p_value <  alpha:
    print("Reject Null Hypothesis")
else:
  print("Fail to Reject Null Hypothesis")

p-value : 0.06680720126885814
Fail to Reject Null Hypothesis


p-value : 0.06680720126885814
Fail to Reject Null Hypothesis p-value < alpha